### PR TITLE
IOPZ-1847: Support multiple K8s roles for single environment

### PR DESCRIPTION
**Pull request by jordannb**
## Testing
1. Ran `./create-k8s-chained-sessions.sh`
2. Confirmed I can connect with or without specifying a persona (`admin`, `dev-writer`, etc.) to connect with (see https://github.com/panorama-ed/school-supplies/pull/167).

### CODE REVIEW

File: create-k8s-chained-sessions.sh [hunk 1]
```diff
@@ -90,7 +96,8 @@ PARENT_SESSION_NAMES="panorama-k8s-playground panorama-k8s-playground-2 panorama
 
 for session in $PARENT_SESSION_NAMES
 do
-    createLeappSession "$session" "AWSAdministratorAccess" "eks-admin-1.24"
-    createLeappSession "$session" "PanoramaK8sEngineeringDefault" "panorama-dev-writer-1.24"
-    createLeappSession "$session" "PanoramaK8sEngineeringDefault" "panorama-dev-reader-1.24"
+    createLeappSession "$session" "AWSAdministratorAccess" "eks" "admin" "1.24"
```
Comment (jordannb) [inline create-k8s-chained-sessions.sh]: Trying to reign in scope creep for now, so I left the `eks` and `panorama` keywords, but only to support the IAM roles without needing to change those too right now.

File: create-k8s-chained-sessions.sh [hunk 2]
```diff
@@ -90,7 +96,8 @@ PARENT_SESSION_NAMES="panorama-k8s-playground panorama-k8s-playground-2 panorama
...
+    createLeappSession "$session" "PanoramaK8sEngineeringDefault" "panorama" "dev-writer" "1.24"
+    createLeappSession "$session" "PanoramaK8sEngineeringDefault" "panorama" "dev-reader" "1.24"
+    createLeappSession "$session" "PanoramaK8sDSAR" "panorama" "dsar-tester" "1.24"
```
Comment (jordannb) [inline create-k8s-chained-sessions.sh:102]: @Pat-Ayres I realized I forgot to update `dsar-tester` to `data-scientist-tester` (or just `data-scientist` if you prefer). I'll hold off on the change and wait for any other code review feedback to minimize back and forth.

Comment (jordannb) [reply create-k8s-chained-sessions.sh:102]: Update: I haven't committed it here yet, but in my testing, I'm using `data-science-tester`, so if no objections to that, I will just stick with that name moving forward.

File: create-k8s-chained-sessions.sh (hunk 1 above)
Comment (Pat-Ayres) [reply create-k8s-chained-sessions.sh]: Those will be changing very soon as part of the in-flight cluster upgrade epic, so I think this makes sense.

File: create-k8s-chained-sessions.sh [hunk 3]
```diff
@@ -15,37 +15,43 @@ declare REGION='us-east-1'
...
 # 1: parent session id name from leapp
 # 2: sso role name to use for the parent session
-# 3: role name to use for the chained session
+# 3: scope of the IAM role ("panorama" or "eks"). used only for legacy reasons
+#    and can be removed when the keyword is removed from the associated IAM
+#    roles.
+# 4: name of the persona (e.g. admin, dev-writer, etc.) the new session is for
+# 5: version of kubernetes, used to determine the associated IAM role
 function createLeappSession {
     green_echo "creating chained session for $1 with role $3"
```
Comment (Pat-Ayres) [inline create-k8s-chained-sessions.sh]: This echo doesn't make sense anymore with the new arguments. It only outputs `panorama` or `eks` instead of a unique role. I think you'll need to do some concatenation to get a role, or just change it to `with persona $4`.

File: create-k8s-chained-sessions.sh [hunk 4]
```diff
@@ -15,37 +15,43 @@ declare REGION='us-east-1'
...
         green_echo "    No parent session found for ${parent_session_name} with role ${parent_role_name}"
         return
     fi
 
-    chained_session_name="${parent_session_name}-${chained_role_name}"
+    chained_session_name="${parent_session_name}-${persona_name}"
 
     green_echo "    looking for existing session ${chained_session_name}"
-    chained_session_id=$(leappSessionId "$chained_session_name" "$chained_role_name")
+    chained_session_id=$(leappSessionId "$chained_session_name" "${persona_name}")
```
Comment (Pat-Ayres) [inline create-k8s-chained-sessions.sh]: This function is broken. The script should be idempotent, but after running it, the second time it still goes into the session creation logic and creates duplicate sessions.

You'll need to pass in the full role name or change the `jq` select function to work as a fuzzy filter.

**Review by Pat-Ayres — CHANGES_REQUESTED** [review]

File: create-k8s-chained-sessions.sh (hunk 3 above)
Comment (jordannb) [reply create-k8s-chained-sessions.sh]: Thank you - I went with the persona here since that's the abstraction people will be abstracting with directly anyways, so I guess it puts the idea out there from the start.

File: create-k8s-chained-sessions.sh (hunk 4 above)
Comment (jordannb) [reply create-k8s-chained-sessions.sh]: This has been fixed now. I retested by re-running the script; output is below (punycode deprecation errors removed manually for readability):
```
% ./create-k8s-chained-sessions.sh                                                                                                                                (~/Development/panorama/leapp-setup) panorama-k8s-playground-admin
creating chained session for panorama-k8s-playground with persona admin
    looking for existing session panorama-k8s-playground-admin
    existing session found
creating chained session for panorama-k8s-playground with persona dev-writer
    looking for existing session panorama-k8s-playground-dev-writer
    existing session found
creating chained session for panorama-k8s-playground with persona dev-reader
    looking for existing session panorama-k8s-playground-dev-reader
    existing session found
creating chained session for panorama-k8s-playground with persona data-science-tester
    looking for existing session panorama-k8s-playground-data-science-tester
    existing session found
creating chained session for panorama-k8s-playground-2 with persona admin
    looking for existing session panorama-k8s-playground-2-admin
    existing session found
creating chained session for panorama-k8s-playground-2 with persona dev-writer
    looking for existing session panorama-k8s-playground-2-dev-writer
    existing session found
creating chained session for panorama-k8s-playground-2 with persona dev-reader
    looking for existing session panorama-k8s-playground-2-dev-reader
    existing session found
creating chained session for panorama-k8s-playground-2 with persona data-science-tester
    looking for existing session panorama-k8s-playground-2-data-science-tester
    existing session found
creating chained session for panorama-k8s-staging with persona admin
    looking for existing session panorama-k8s-staging-admin
    existing session found
creating chained session for panorama-k8s-staging with persona dev-writer
    looking for existing session panorama-k8s-staging-dev-writer
    existing session found
creating chained session for panorama-k8s-staging with persona dev-reader
    looking for existing session panorama-k8s-staging-dev-reader
    existing session found
creating chained session for panorama-k8s-staging with persona data-science-tester
    looking for existing session panorama-k8s-staging-data-science-tester
    no existing session found; starting session for panorama-k8s-staging to get role arn
An error occurred (NoSuchEntity) when calling the GetRole operation: The role with name panorama-data-science-tester-1.24 cannot be found.
    creating new profile
    creating new session
creating chained session for panorama-k8s-production with persona admin
    looking for existing session panorama-k8s-production-admin
    existing session found
creating chained session for panorama-k8s-production with persona dev-writer
    looking for existing session panorama-k8s-production-dev-writer
    existing session found
creating chained session for panorama-k8s-production with persona dev-reader
    looking for existing session panorama-k8s-production-dev-reader
    existing session found
creating chained session for panorama-k8s-production with persona data-science-tester
An error occurred (NoSuchEntity) when calling the GetRole operation: The role with name panorama-data-science-tester-1.24 cannot be found.
```

Comment (Pat-Ayres) [reply create-k8s-chained-sessions.sh]: (fyi, you can get rid of the punycode output by changing your host installed node version. I opened [a ticket](https://github.com/Noovolari/leapp/issues/500) with the leapp team about it with the workaround I used described.

**Review by Pat-Ayres — APPROVED** [review]

**Review by Pat-Ayres — APPROVED** [review]
